### ISSUE TITLE
I would like my enum input values to be the enum instance instead of the enum values

### ISSUE THREAD

**Issue by radix** [graphql-python/graphene#1151]
Is there a way for me to do this?

Here is some example code.

```python
from enum import Enum, auto

from graphene import Enum as GQLEnum, ObjectType, Schema, String
from graphene.relay import ClientIDMutation
from graphene.test import Client

class EnumThing(Enum):
    a = auto()
    b = auto()

GQLEnumThing = GQLEnum.from_enum(EnumThing)

class TestMut(ClientIDMutation):
    class Input:
        enumthing = GQLEnumThing(required=True)

    enumtype = String()

    @classmethod
    def mutate_and_get_payload(cls, root, info, enumthing, client_mutation_id=None):
        print("enumthing is", repr(enumthing), type(enumthing))
        return cls(enumtype=type(enumthing).__name__)

class Mutations(ObjectType):
    testmut = TestMut.Field()

schema = Schema(mutation=Mutations, auto_camelcase=False)

client = Client(schema)

mutation = '''
mutation whatever {
    testmut(input: {enumthing: a}) {
        enumtype
    }
}
'''

print(client.execute(mutation))
```

When I run this, I get the following output:

```
enumthing is 1 <class 'int'>
{'data': OrderedDict([('testmut', OrderedDict([('enumtype', 'int')]))])}
```

Instead of getting the integer `1` passed to my mutation function, I would like to have `EnumThing.a` passed, which is an instance of `EnumThing`. I haven't figured out where in graphene this translation of the literal `a` to the value `1` is actually happening (I would expect an access of the `.value` attribute on the enum somewhere).

Why? because I don't really care about the integer `1` -- that's just something generated by Python. If I log the value of this enum, I want to see that it's a `EnumThing.a`, not the integer `1`. If I pass this thing to the rest of my codebase which is expecting it to be an instance of `EnumThing`, it breaks. So I end up converting it *back* to the instance from the integer that Graphene gave me.

**Comment by jkimbo** [comment]
@radix yep this is a change that we are going to make as part of v3. I've created a PR for it here: https://github.com/graphql-python/graphene/pull/1153

**Comment by tymokvo** [comment]
Just upping this request. Thanks for looking into it!

**Comment by skewty** [comment]
if the `int` value isn't important and you want a human friendly way to specify the value there is always `str` values.

```py
from enum import Enum, auto

__all__ = ['StrEnum', 'auto']

class StrEnum(Enum):
    """Auto generates unique lowercase str names for values.

    Allows quick enum generation code like:
       Foo = StrEnum("Foo", "CREATED ADDED REMOVED")
    which rougly equates to:
        class Foo(Enum):
           CREATED = "created"
           ADDED = "added"
           REMOVED = "removed"
    or
        class Foo(StrEnum):
           CREATED = auto()
           ADDED = auto()
           REMOVED = auto()
   """

    # noinspection PyMethodParameters
    def _generate_next_value_(name, start, count, last_values) -> str:
        return name.lower()

    def __eq__(self, other: object) -> bool:
        return self.value.__eq__(other)

    def __ne__(self, other: object) -> bool:
        return self.value.__ne__(other)

    def __hash__(self) -> int:
        return self.value.__hash__()

    def __str__(self) -> str:
        return self.value

    def __len__(self) -> int:
        return self.value.__len__()

    def __cmp__(self, o: object):
        return self.value.__cmp__(o)

    def __gt__(self, other: object) -> bool:
        return self.value.__gt__(other)

    def __ge__(self, other: object) -> bool:
        return self.value.__ge__(other)

    def __lt__(self, other: object) -> bool:
        return self.value.__lt__(other)

    def __le__(self, other: object) -> bool:
        return self.value.__le__(other)
```

**Comment by jkimbo** [comment]
This was resolved in #1153
